### ISSUE TITLE
option to stop vpnhotspot if vpn disconnects

### ISSUE THREAD

**Issue by magixtruct** [Mygod/VPNHotspot#177]
hi @mygod,

can you add an option to automatically stop vpnhotspot + android hotspot if the vpn disconnects?
in my case the hotspot was working on after that (connected to a windows device) + began leaking the real ip.
can send firewall logs if you want.
I assume you are aware of that, but haven't added it cause of convenience for the user (whilst gaming etc.)
but as an optional setting for more security? would make vpnhotspot one step closer to perfect imo.

apart from that, vpn hotspot gets better + better with every update. I have big respect for your work + your genius. just sayin'. 😃

**Comment by Mygod** [comment]
Set fallback network to `none` to prevent IP leaks.

**Comment by magixtruct** [comment]
had set fallback network to none.
problem here is, that vpn hotspot should work as a firewall, preventing the hotspot to connect to the internet if not connected to vpn, imo.

**Comment by QuickOK** [comment]
Are you only worried about hotspot clients or your hotspot phone and
client? If all of the above you need to setup Always-On VPN on android and
"block connections without VPN".

**Comment by magixtruct** [comment]
had set always on vpn + block connections without vpn. I always do this.

**Comment by QuickOK** [comment]
I asked because it theoretically should have all connections blocked. Is it
not the case?

**Comment by magixtruct** [comment]
vpns, even if always on vpn + block connections without vpn, just block the normal inet connection, not the hotspot connection. that's a seperate. that's an important reason why apps like vpnhotspot exist imo. maybe there are very few vpns/vpn clients that can do this, maybe strong swan or wireguard, dunno. but not one of the vpns I've used + they're pretty good ones: windscribe + proton vpn. I even blocked lan connections in windscribe. but no effect on the hotspot. it's an android thing. they route the hotspot traffic seperately. my phone is samsung android 8.1.0. one thing I have to admit: I use tethering hardware acceleration + I want to keep it. I like it a lot.
but there's another vpn tether app on playstore (maybe a fork of this one), that works as a proxy + has options to prevent the hotspot from connecting, that works fine for me.
I just thought, if vpnhotspot would have this feature too, that'd be great!

**Comment by Mygod** [comment]
The default setting in this app does not block non-VPN connections. To do that, enable "disable IPv6 tethering" and set fallback network to `none`. If you still observe connection leaks please open a separate issue.

**Comment by magixtruct** [comment]
ipv6 tethering was disabled + had set fallback network to `none` at the time I wrote the last issue. but I have reproduced the issue right now to be absolutely sure. but it's fixed since latest version! tyvm! also overall app experience is even more better now.

if additionally the hotspot could be prevented to connect, before vpnhotspot app suddenly stops or when vpnhotspot app suddenly stops, that'd make the whole thing absolutely perfect.🤩
